src/libexpr/value.hh

```diff
@@ -447,6 +442,19 @@ public:
 };
 
 
+extern Value prim_blackHole;
+
+inline bool Value::isBlackhole() const
+{
+    return internalType == tApp && app.left == &prim_blackHole;
+}
+
+inline void Value::mkBlackhole()
+{
+    mkApp(&prim_blackHole, &prim_blackHole);
```

Comment:
May be sufficient to only set the type and `left`, but haven't verified. Already ok as is.